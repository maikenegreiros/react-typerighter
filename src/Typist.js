import React from "react";
import PropTypes from "prop-types";
import styled, { keyframes } from "styled-components";

const steps = 50;

const blinkCursor = keyframes`
  from {
    border-color: transparent;
  }
  to {
    border-color: inherit;
  }
`;

const typing = keyframes`
  from {
    width: 0;
  }
  to {
    width: 100%;
  }
`;

const LineContainer = styled.div`
  display: inline-block;
`;

const Line = styled.p`
  display: inline-block;
  color: ${(props) => props.color};
  font-family: ${(props) => props.fontFamily};
  font-size: ${(props) => props.fontSize};
  border-right: 2px solid ${(props) => props.color};
  animation: ${typing} ${(props) => props.typingTime}s steps(${steps}) normal,
    ${blinkCursor} ${(props) => props.blinkTime}s steps(${steps}) infinite
      normal;
  overflow: hidden;
  white-space: nowrap;
`;

function Typist({ text, color, fontSize, fontFamily, blinkTime, typingTime }) {
  return (
    <LineContainer>
      <Line
        color={color}
        fontSize={fontSize}
        fontFamily={fontFamily}
        blinkTime={blinkTime}
        typingTime={typingTime}
      >
        {text}
      </Line>
    </LineContainer>
  );
}

Typist.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string,
  fontSize: PropTypes.string,
  fontFamily: PropTypes.string,
  blinkTime: PropTypes.number,
  typingTime: PropTypes.number,
};

Typist.defaultProps = {
  color: "#868686",
  fontSize: "1em",
  fontFamily: "cursive",
  blinkTime: 1,
  typingTime: 4,
};

export default Typist;
