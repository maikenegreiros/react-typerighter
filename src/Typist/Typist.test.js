import React from "react";
import { shallow } from "enzyme";
import Typist from "./Typist";

describe("Typist", () => {
  it("should render with passed text", () => {
    const text = "Text will be here";
    const component = shallow(<Typist text={text} />);
    expect(component.text()).toBe(text);
    component.unmount();
  });

  it("should throw an error by not passing text prop", () => {
    const componentMaker = () => shallow(<Typist />);
    expect(componentMaker).toThrowError();
  });

  it("should render matching snapshot", () => {
    const component = shallow(
      <Typist
        text="Hello there!"
        color="#868686"
        fontSize="2em"
        fontFamily="monospace"
        blinkTime={3}
        typingTime={10}
      />
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
