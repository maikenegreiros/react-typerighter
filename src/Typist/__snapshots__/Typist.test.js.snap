// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typist should render matching snapshot 1`] = `
<styled.div>
  <styled.p
    blinkTime={3}
    color="#868686"
    fontFamily="monospace"
    fontSize="2em"
    typingTime={10}
  >
    Hello there!
  </styled.p>
</styled.div>
`;
